### PR TITLE
Fix/terminals

### LINKED ISSUE #2: Fix in-slide terminals not connecting
We have to migrate the in-slide terminals from an iframe approach to a single-websocket direct approach which requires altering of the backend. This is because it is only allowed to open 2 websockets at once on one domain in this browser.

### PR DESCRIPTION
**Pull request by martin31821**
Depends on #8.
Fixes #2.

### CODE REVIEW

File: src/app/slides/slides.component.html [hunk 1]
```diff
@@ -121,8 +121,8 @@ <h3>
       <div class="layout-center-top">
         Configure your environment!
         <pre>$ git config --global user.name "name"
-  $ git config --global user.email "email"
-  $ git config --global core.editor vim</pre>
+$ git config --global user.email "email"
```
Comment (fin-ger) [inline src/app/slides/slides.component.html]: Please remove these whitespaces

File: src/app/slides/slides.component.html [hunk 2]
```diff
@@ -121,8 +121,8 @@ <h3>
...
+$ git config --global core.editor vim</pre>
```
Comment (fin-ger) [inline src/app/slides/slides.component.html]: Please remove these whitespaces

File: src/app/tty/tty.component.ts [hunk 3]
```diff
@@ -1,26 +1,43 @@
...
 @Component({
   selector: 'tty',
   templateUrl: './tty.component.html',
   styleUrls: ['./tty.component.scss']
 })
-export class TtyComponent implements OnInit {
-  @Input('port') public port: string;
+export class TtyComponent implements AfterViewInit {
+  @ViewChild('terminal')
+  private terminalElement: ElementRef;
```
Comment (fin-ger) [inline src/app/tty/tty.component.ts]: Add underscore prefix for private members

File: src/app/tty/tty.component.ts [hunk 4]
```diff
@@ -1,26 +1,43 @@
...
+  private terminal: Terminal;
```
Comment (fin-ger) [inline src/app/tty/tty.component.ts]: Add underscore prefix for private members.

**Comment by martin31821** [comment]
Server instructions can be obtained [here](https://github.com/ovgu-cs-workshops/git-composefiles).
Changing the font size has unintended side effects I can't explain or fix, so you have to look into it yourself.

**Comment by fin-ger** [comment]
Ok thanks! Approving this PR then

**Pull request merged**